File: Frontend/src/types/contract.ts
// types/contract.ts

// 1. Payload để tạo mới (Khớp 100% với JSON bạn đưa)
export interface CreateContractParams {
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    startDate: string; // Định dạng ISO: "2025-12-13T02:14:56.688Z"
    endDate: string;   // Định dạng ISO
    companyName: string;
    bankAccountNumber: string;
    pdfLink: string;
    resellerId: number;
    addressId: number;
}

// 2. DTO trả về khi Get Detail hoặc List (Thường backend sẽ trả về thêm ID và ContractNumber)
export interface ContractDto {
    id: number;
    contractNumber: string; // Backend tự sinh
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    startDate: string;
    endDate: string;
    companyName: string;
    bankAccountNumber: string;
    pdfLink: string;
    resellerId: number;
    addressId: number;
    // Các field bổ sung nếu có relation
    reseller?: { id: number; name: string };
    address?: { id: number; street: string; city: string };
}

// 3. Query Params cho Filter (Giữ nguyên logic cũ)
export interface ContractQueryParams {
    search?: string;
    resellerId?: number;
    startDateFrom?: string;
    startDateTo?: string;
    pageNumber: number;
    pageSize: number;
    sortBy?: string;
    sortDesc?: boolean;
}

// 4. Kết quả phân trang
export interface PagedResult<T> {
    items: T[];
    totalCount: number;
    pageNumber: number;
    pageSize: number;
    totalPages: number;
}
export interface UpdateContractParams extends CreateContractParams {
    id: number;
}

File: Backend/src/CustomerService/Domain/Entities/Contract.cs
using Domain.Common;

namespace Domain.Entities;

public class Contract : BaseEntity
{
    public string ContractNumber { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }

    // --- Thông tin khách hàng ---
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    
    // --- Thông tin bổ sung ---
    public string? CompanyName { get; set; }
    public string? BankAccountNumber { get; set; }
    public string? PdfLink { get; set; } // Link file PDF từ S3 trả về

    // --- Các mối quan hệ (Foreign Keys) ---
    
    // 1. Quan hệ với Reseller (Nhiều Contract thuộc về 1 Reseller)
    public int ResellerId { get; set; }
    public Reseller Reseller { get; set; } = null!;

    // 2. Quan hệ với Address (1 Contract có 1 Address)
    public int AddressId { get; set; }
    public Address Address { get; set; } = null!;

    // 3. Quan hệ với Order (1 Contract có nhiều Order)
    public ICollection<Order> Orders { get; set; } = new List<Order>();

    // 4. Quan hệ với History (1 Contract có nhiều lịch sử thay đổi)
    public ICollection<ContractHistory> History { get; set; } = new List<ContractHistory>();
}